### DISCUSSION TITLE
How to assert sub action was run when manually setting job connection?

### DISCUSSION THREAD

**Discussion by edalzell** [lorisleiva/laravel-actions#209] [Q&A]
Hi there,

I have an action where I manually set the connection:

```php
use AsAction;

public string $jobConnection = 'redis';

public function handle(Orphan $orphan)
```

And it's called from a model like this:

```php
SendOrphanAgeOutNotification::dispatch($this);
```

In my test I have I have `<env name="QUEUE_CONNECTION" value="sync"/>` but I do have reds configured and running. Test is:

```php
Queue::fake();

SendOrphanAgeOutNotifications::run();
SendOrphanAgeOutNotification::assertPushedOn('default');
```

But that test never passes. I know this code works because I've removed that `$jobConnection` and run other tests.

What am I doing wrong?

**Comment by lorisleiva** [comment]
Hi there 👋 

I'm a bit confused. It looks like you're calling `dispatch` in your source code to dispatch a job but `run` in your tests which will run the class immediately as an object (therefore, not dispatching a job in any queue).

What are you trying to test exactly?

**Reply by edalzell** [reply]
Ya there's some subtlety here. Note that there's an action w/ an S (SendOrphanAgeOutNotifications) that I'm calling `run` on. It's normally called as a command.

<img width="897" alt="CleanShot 2022-11-10 at 09 36 50@2x" src="https://user-images.githubusercontent.com/6069653/201166922-ca6e2dc0-228d-45d4-8048-1606adfab3b6.png">

It ends up calling `SendOrphanAgeOutNotification` via `dispatch` and that action sends an email:

<img width="651" alt="CleanShot 2022-11-10 at 09 35 24@2x" src="https://user-images.githubusercontent.com/6069653/201166621-f5fd495f-2e0c-4d8f-93fe-feb43ccff30b.png">

<img width="1044" alt="CleanShot 2022-11-10 at 09 34 53@2x" src="https://user-images.githubusercontent.com/6069653/201166558-01e89e22-192c-4d4c-857a-8af5f17afc0b.png">

So now I'd like to test that that action (`SendOrphanAgeOutNotification`) is actually queued. But the test code above doesn't work, so clearly I'm doing something wrong.
